### ISSUE TITLE
Using Huggingface Trainer in Colab -> Disk Full

### ISSUE THREAD

**Issue by k-for-code** [huggingface/transformers#14922]
https://discuss.huggingface.co/t/using-huggingface-trainer-in-colab-disk-full/5951/2 

I faced the same issue mentioned in this discussion. Please give it a look.

[## Environment info
- `transformers` version: 4.15.0
- Platform: Linux-5.4.144+-x86_64-with-Ubuntu-18.04-bionic
- Python version: 3.7.12
- PyTorch version (GPU?): 1.10.0+cu111 (True)
- Tensorflow version (GPU?): 2.7.0 (True)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no

### Who can help

- Trainer: @sgugger

## Information
This issue is model agnostic
I believe it's task agnostic too

## To reproduce

1. Run this notebook in colab
 https://colab.research.google.com/github/huggingface/blog/blob/master/notebooks/01_how_to_train.ipynb#scrollTo=YpvnFFmZJD-N

## Expected behavior
1. Disk should not go out of space,  Because I used `save_total_limit=1`  in the training argument. So previous checkpoints should be deleted

**Comment by k-for-code** [comment]
I guess I found the reason, 
on deleting the previous checkpoint, it goes to the google drive bin and the bin does not delete it then (deletes after 30 days) and this results in occupied space. 
One way to solve this will be overriding the `_rotate_checkpoint` method of the `trainer` to also clean the drive bin, but this would require authentication from the user every time to empty the bin.
Any idea around implementing the same?

**Comment by sgugger** [comment]
Looks like an issue on Colab that they don't let the user choose to delete things without going in that bin (or clear that bin when space is needed).

The only workaround I can think of is to use a very high value as a saving step, or to disable the saving altogether during training with `save_strategy="no"`.

**Comment by OneCodeToRuleThemAll** [comment]
> 

Still facing the same issue.
+Having save_strategy = 'no' means that we are not able to pick the best model based on our evaluation metric :(

**Comment by sgugger** [comment]
You can raise the issue on Colab. There is really nothing we can do on our side since the problem is that Colab does not let us delete older checkpoints.
